monkestation/code/game/objects/items/melee/armaments.dm

```diff
@@ -0,0 +1,146 @@
+/obj/item/armament
+	name = "armament"
+	desc = "a daemon weapon."
+	force = 30
+	throwforce = 10
+	w_class = WEIGHT_CLASS_BULKY
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	icon = 'monkestation/icons/obj/items_and_weapons.dmi'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "tore", "ripped", "diced", "cut")
+	slot_flags = ITEM_SLOT_BACK|ITEM_SLOT_BELT
+	sharpness = IS_SHARP
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF | INDESTRUCTIBLE
+	var/last_ability_use = 0 // time the last ability was used
+	var/ability_cooldown = 5 SECONDS
+
+/obj/item/armament/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(last_ability_use + ability_cooldown > world.time )
+		return TRUE
+	last_ability_use = world.time
+
+/obj/item/armament/daemon_blade
+	name = "daemon blade"
+	desc = "A blade formed out of demonic energy. Activate in hand to throw flames."
+	icon_state = "daemon_blade"
+	item_state = "daemon_blade"
+	lefthand_file = 'monkestation/icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'monkestation/icons/mob/inhands/weapons/swords_righthand.dmi'
+	force = 12
+	ability_cooldown = 20 SECONDS
+	var/flame_toggle = FALSE
+
+/obj/item/armament/daemon_blade/Initialize(mapload)
+	. = ..()
+
+/obj/item/armament/daemon_blade/attack_self(mob/user)
+	. = ..()
+	if(last_ability_use + ability_cooldown > world.time ) //Can't turn it on if its on cooldown
+		flame_toggle = FALSE
+		return
+	flame_toggle = !flame_toggle
+	if (flame_toggle)
+		icon_state = "daemon_blade_fire"
+		to_chat(user, "<span class='warning'>[src] glows bright with mysterious runes. Click a target to burn it</span>")
+	else
+		icon_state = "daemon_blade"
+		to_chat(user, "<span class='warning'>[src] stops glowing.</span>")
+	return
+
+/obj/item/armament/daemon_blade/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(flame_toggle)
+		. = ..()
+		var/turflist = get_line(user, get_turf(target))
+		flame_turf(turflist)
+		flame_toggle = FALSE
+		icon_state = "daemon_blade"
+		to_chat(user, "<span class='warning'>[src] stops glowing.</span>")
+
+/obj/item/armament/daemon_blade/proc/flame_turf(turflist)
+	var/turf/previous_turf = get_turf(src)
+	for(var/turf/current_turf in turflist)
+		if(current_turf == get_turf(src))
+			continue
+		if(previous_turf == get_turf(src))
+			previous_turf = current_turf
+			continue
+		if(isclosedturf(previous_turf)) //Can't fire fire through walls
```

Comment:
You'll want a check like this for windows assuming you don't want it flying through windows or airlocks. as they are considered structures and not closed turfs

Reply:
If it should be changed anyway lmk but I think I like it going through airlocks and windows.